ci/deps/actions-39.yaml

```diff
@@ -12,11 +12,30 @@ dependencies:
   - hypothesis>=5.5.3
 
   # pandas dependencies
+  - beautifulsoup4
+  - bottleneck
+  - fsspec>=0.8.0, <2021.6.0
+  - gcsfs
+  - html5lib
+  - jinja2
+  - lxml
+  - matplotlib
+  - moto>=1.3.14
+  - flask
+  - numexpr
   - numpy
+  - openpyxl
+  - pyarrow
+  - pytables
   - python-dateutil
   - pytz
-
-  # optional dependencies
-  - pytables
+  - s3fs>=0.4.2
   - scipy
-  - pyarrow=1.0
```

Comment:
This is to test against the min version of pyarrow that the pyarrow backed string array supports. backporting this change will still leave `pyarrow=1.0.0` in `ci/deps/actions-38-locale.yaml` on 1.3.x

Reply:
pyarrow min is supposed to be bumped to 1.0 in 1.4 if I'm reading #41329 right, so we can probably pin in the minimum versions CI(3.8). 

For 1.3.x, I think its fine to leave this out.
P.S. Thanks for the backport by the way.